### DISCUSSION TITLE
Button Logs for later use

### DISCUSSION THREAD

**Discussion by navdeepssidhu** [arendst/Tasmota#16917] [General]
Hi, 
I am running Tasmota 12.2 on ESP8266 and it is running fine. I want to know 
a) Is it possible to save Button on/off logs to text file or  some other way to save logs so that i can check appliance was on / off duration and timings. 
b) How can i convert %Time% (minutes since midnight) to Hours and minutes and store in variable
any help is appreciated

**Comment by sfromis** [comment]
If you want to check appliance on/off, do not look at button status, but instead power status. By default Tasmota will publish power status changes via MQTT, and then you can have a client capture data, and to what you want. You could also capture the log messages via serial, syslog, MQTT log (which is an additional feature) or shorter term just the web console. You can set logging level for more details, like when log level 3 does include button actions, if you really want that too.

Data manipulation like converting to hh:mm is ideally not done on Tasmota, but of course it is possible to do the calculations in a somewhat laborious way, in rules or scripting. My preference would be to avoid this being a "need". With local processing needs, my recommendation would be ESP32 and Berry.

**Reply by navdeepssidhu** [reply]
Thanks for taking time to address my question.  As suggested i can capture Power status via MQTT when connected, my device works as standalone most of the time, and when it is connected to MQTT i want to check whether appliances ran as per schedule or not, as some time power failure disturb the device, also connecting to a battery is not a option for me. This is the reason i am looking to save Power logs to Text file in LittleFS or alternate option which can be retrieved later. Of course i can save logs with time stamp to Mem, but it is very limited.

**Reply by sfromis** [reply]
Tasmota is just not designed for unconnected operation, or local data aggregation, the "M" in the name is for MQTT. To have data saved locally in the file system, you'll have to code it yourself, likely using scripting. Takes some learning curve. Easier with ESP32 and Berry.

BTW. Without network connection, you cannot rely on having local time available, without some other real time source.

**Reply by navdeepssidhu** [reply]
I am not an advance user, right now scripting/Berry is quite a big task for me. Can you please guide me how to convert Minutes since midnight (%Time%) to  hours, i can successfully get the minutes and to save to Var2 with rules 
rule1 ON Power1#state DO  backlog var2=%Time%%60  endon
but struggling to get Hours.
